test/common/http/http2/conn_pool_test.cc

```diff
@@ -288,11 +288,27 @@ TEST_F(Http2ConnPoolImplTest, DrainConnectionConnecting) {
   pool_.drainConnections();
 
   // Cancel the pending request, and then the connection can be closed.
-  r.handle_->cancel();
+  r.handle_->cancel(Envoy::ConnectionPool::CancelPolicy::Default);
   EXPECT_CALL(*this, onClientDestroy());
   pool_.drainConnections();
 }
 
+/**
+ * Verify that on CloseExcess, the connection is destroyed immediately.
+ */
+TEST_F(Http2ConnPoolImplTest, CloseExcess) {
```

Comment:
I think this test should exercise the logic for `connecting_request_capacity_` a bit more by setting the max concurrent requests to 2, starting 4 pending requests, then cancelling all 4 and EXPECTing a closed requests on every other cancel.